Resolver/EntityAnnotationResolver.php

```diff
@@ -218,6 +218,11 @@ public function evaluateMapping(AnnotationEntity $annotation, $entity)
 
             if (!($instance instanceof $entityClass)) {
 
+                $notFoundException = $annotation->getNotFoundException();
+                if (!empty($notFoundException)) {
+                    throw new $notFoundException['exception']($notFoundException['message']);
```

Comment:
@rogergros Meeeec! Remember that the minimum version of PHP for using this package is `5.3.3`, and I think that this feature is still not available in `5.3.3`...